Event manager3

Jokardoo opened the pull request:
Iteration3

Jokardoo committed:
Iteration3

Jokardoo committed:
added edits to the iteration3

Jokardoo committed:
added edits to the third iteration

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/controller/EventController.java:
```diff
@@ -0,0 +1,108 @@
+
+        Event newEvent = new Event();
+
+        newEvent.setLocationId(request.getLocationId());
```
Нужно написать маппер для EventCreateRequestDto в Event

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventService.java:
```diff
@@ -0,0 +1,235 @@
+        checkEventFields(event);
+        // All right
+
+        EventDto eventDto = new EventDto();
```
Здесь нет необходимости в этом DTO. Смотри на каждом слое приложения мы работаем со своим типом обьекта на контроллере работаем с DTO в данном случае с EventCreateRequestDto, на сервисном слое с бизнес объектом в данном случае с Event, на слое с репозиторием работаем с Entity в данном случае с EventEntity. В твоём случае у нас получилось что на контроллере мы работаем с EventCreateRequestDto мапим в Event потом на сервисной логике из Event мапим в EventDto потом обратно в Event и дальше уже всё как нужно у тебя тебе из этой цепочки нужно убрать маппинг из Event в EventDto. Пересмотри все такие места

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventService.java:
```diff
@@ -0,0 +1,235 @@
+                logger.warn("WARN: Incorrect date exception!");
+                throw new IncorrectDateException("The date of the event cannot be in the past tense!");
+            }
+        } catch (Exception e) {
```
Не корректно через catch делать валидации. И я не совсем понимаю что у тебя его вызовет сравниваешь ты одинаковый тип данных

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/controller/EventController.java:
```diff
@@ -0,0 +1,108 @@
+
+        Event event = new Event();
+
+        event.setName(eventUpdateRequestDto.getName());
```
Тут тоже нужно описать маппер из EventUpdateRequestDto в Event и использовать его

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventService.java:
```diff
@@ -0,0 +1,235 @@
+    }
+
+    public List<Event> getFilteredEvents(EventSearchRequestDto requestDto) {
+        List<EventEntity> entityList = eventRepository.searchByFilter(
```
Тут не использовал бизнес обьект а сразу из DTO работаешь с Entity

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+
+import jokardoo.eventmanager.domain.event.Event;
+import jokardoo.eventmanager.domain.event.EventStatus;
+import jokardoo.eventmanager.domain.event.eventRegistration.EventRegistration;
```
Нет такого обьекта в проекте, аналогично и с EventRegistrationEntity

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+
+        EventRegistrationEntity createdRegistration = eventRegistrationRepository.save(eventRegistrationMapper.modelToEntity(eventRegistration));
+        event.setOccupiedPlaces(event.getOccupiedPlaces() + 1);
+        System.out.println(event.getOccupiedPlaces());
```
Для логирования нужно использовать логер т.к принты будут видны только локально в консоли

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+    public EventRegistration registerUserOnEvent(Long eventId) {
+        Long userId = authParser.getId();
+
+        EventRegistration eventRegistration = new EventRegistration();
```
Если сделал бизнес объект то и дальше работай через него, а не подставляй параметры на прямую. Ниже в этом методе отмечу где

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+        eventRegistration.setEventId(eventId);
+        eventRegistration.setUserId(authParser.getId());
+
+        Event event = eventService.getById(eventId);
```
как здесь

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+        Event event = eventService.getById(eventId);
+        //check
+
+        if (eventRegistrationRepository.existsByEventIdAndUserId(eventId, userId)) {
```
и здесь

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventRegistrationService.java:
```diff
@@ -0,0 +1,79 @@
+
+    private final AuthenticationParser authParser;
+
+    public EventRegistration registerUserOnEvent(Long eventId) {
```
А для чего нам здесь возвращаемое значение если мы его потом нигде не используем

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/bean/SchedulerConfig.java:
```diff
@@ -0,0 +1,13 @@
+@EnableAsync
+@Configuration
+@ConditionalOnProperty(name = "scheduler.enabled", matchIfMissing = true)
+public class SchedulerConfig {
```
Всю логику расписания вынеси в этот класс и переименуй его в EventScheduler и я бы его вынес в отдельный пакет scheduler. И для чего тебе нужно использование аннотаций @Async в расписании?

sorokinuniversity commented on src/main/resources/application.properties:
```diff
@@ -15,3 +15,7 @@ jwt.expiration=86400000
 
 server.port=8080
+
+spring.kafka.bootstrap-servers=localhost:9092
```
Убирай из ПР всё что не относится к текущему заданию. Потом добавишь в новый ПР если это какие-то правки необходимые по предыдущим ПР то ок

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/bean/MapperBeanConfiguration.java:
```diff
@@ -0,0 +1,14 @@
+public class MapperBeanConfiguration {
+    @Bean
+    public EventLocationMapper eventLocationMapper() {
+        return new EventLocationMapper();
```
Смотри ты ведь можешь пометить этот класс EventLocationMapper аннотацией @Component и внедрять бин этого класса где тебе это необходимо

Jokardoo committed:
Iteration3

Jokardoo committed:
Iteration3

Jokardoo committed:
Iteration3

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/controller/EventController.java:
```diff
@@ -0,0 +1,103 @@
+
+    private final RegistrationService registrationService;
+
+    private final EventCreateRequestMapper eventCreateRequestMapper;
```
Нет классов в проекте EventCreateRequestMapper, EventUpdateRequestMapper, EventSearchRequestMapper

Jokardoo committed:
Add class: EventSearchRequest,

Add mappers: EventCreateRequestMapper,
EventSearchRequestMapper, EventUpdateRequestMapper

Jokardoo committed:
Add classes: Registration, RegistrationEntity

Jokardoo committed:
Добавил недостающие классы, заменил мапперы на реализацию mapstruct

Jokardoo committed:
Removed kafka dependency from pom.xml

Jokardoo committed:
-

Jokardoo committed:
delete kafka property from application.properties

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/service/EventService.java:
```diff
@@ -0,0 +1,223 @@
+
+    private final EventLocationService eventLocationService;
+
+    private final EventChangesSender eventChangesSender;
```
Все классы связанные не с текущей задачей, а с последующими задачами необходимо убрать из этого мержа

Jokardoo committed:
update EventService

Jokardoo committed:
delete class UserCredentials

Jokardoo committed:
refactoring EventService

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/scheduler/EventScheduler.java:
```diff
@@ -0,0 +1,42 @@
+    public void updateEventsStatus() {
+        LocalDateTime now = LocalDateTime.now();
+
+        List<Event> eventList = eventService.findAll();
```
На данном этапе ок. Но на будущее тут лучше писать выборку с условием из бд чтобы у тебя было меньше выборка с которой нужно выполнить работу соответственно ты быстрее её выполнишь и второе у тебя будет меньше нагрузка на приложение.

sorokinuniversity commented on src/main/java/jokardoo/eventmanager/security/jwt/JwtAuthenticationService.java:
```diff
@@ -30,7 +29,6 @@ public String authenticateUser(SignUpRequest signUpRequest) {
                 )
         );
 
-        // TEST AUTHENTICATION
```
Мы вроде ещё в предыдущем пр обсуждали этот момент тут у тебя 2 проверки пользователя одна spring которая выполняется в методе authenticate, а другая которую ты сам делаешь и возвращаешь сгенерированный токен. По хорошему тебе чтобы запрос не делал двойную работу оставить одну из них и возвращать сгенерированный токен
